integration/rpcserver_test.go

```diff
@@ -94,10 +95,80 @@ func testGetBlockHash(r *rpctest.Harness, t *testing.T) {
 	}
 }
 
+func testMiningAddrCalls(r *rpctest.Harness, t *testing.T) {
+	var err error
+	const numAddrs = 5
+
+	// First generate a few fresh addresses from the harness.
+	addrs := make([]btcutil.Address, numAddrs)
+	for i := 0; i < numAddrs; i++ {
+		addrs[i], err = r.NewAddress()
+		if err != nil {
+			t.Fatalf("unable to generate new address: %v", err)
+		}
+	}
+
+	// Next, we use the addminingaddr call to add the above generated
+	// addresses to btcd's set of mining addresses.
+	for _, addr := range addrs {
+		if err := r.Node.AddMiningAddr(addr); err != nil {
+			t.Fatalf("unable to add mining address: %v", err)
+		}
+	}
+
+	// Attempting to add the same address twice should be rejected due to
+	// duplicates.
+	if err := r.Node.AddMiningAddr(addrs[0]); err == nil {
+		t.Fatalf("duplicate mining address wasn't rejected")
+	}
+
+	// Next, try to delete a single address, then immediately try to delete
+	// the same address. The second call should result in an error as the
+	// address is no longer within the set of mining addresses.
+	if err := r.Node.DelMiningAddr(addrs[0]); err != nil {
```

Comment:
Excellent, point. I've changed the duplicate deletion detection portion in the integration test in order to exercise deletion from near the mid-point of the slice.